This notebook is the first in a series of jupyter notebooks intended to introduce students to programing in python. I designed it for the Vaughan house Living Learning Community at UC Merced. It focuses on basic math functions in python 3, and introduces students to the concept of variables.

We're going to start with some simple math. First, addition and subtraction. In the cells below I'm doing some simple addition and subtraction. A useful thing to know about python is that any line that starts with a "#" will be ignored by python as it evaluates a cell. This is what we refer to as a comment, and can be used to both describe the code you are writing, and removing lines from evaluation when you are trying to fix errors ("debugging").

In [1]:
# what's 3 + 7?
3+7

10

In [2]:
# what's 3 - 7?
3-7

-4

In [3]:
# what's 13 - 3 + 5 + 1 - 8?
13-3+5+1-8

8

In [4]:
# what's 21-4?
21 - 4

# what's 17 + 31?
17 + 31

48

You'll notice that the last cell, above, only outputs the second math line (17+31), and we don't see the output from the first math line (21-4). Jupyter will only show the output for the last line in a cell, even though it evaluates the entire cell. If we want to see the output for a line that isn't the last line, we need to use the print() function:

In [5]:
# what's 21-4?
print(21-4)

# what's 17 + 31?
print(17+31)


17
48


Now, here's some multiplication and division.

In [6]:
# what's 3x8?
3 * 8

24

In [7]:
# what's 17/4?
17 / 4

4.25

In [8]:
# what's 22/7?
22 / 7

3.142857142857143

Because $\frac{22}{7}$ is irrational, it will continue forever, so python cuts it off after 15 digits. However, I can use another built-in function, the format() function, combined with the print() to change how many decimals are shown.

In [14]:
# what is 22/7, keeping just 3 digits?
print("{:.3}".format(22/7))

# what is 22/7, keeping 5 digits?
print("{:.5}".format(22/7))

3.14
3.1429


The format() function will also let me change how the number is shown in other ways. For example, I can use it to show a number in scientific notation by adding an 'E' inside the braces.

In [17]:
# what is 179/7, keeping 5 digits?
print("{:.4E}".format(179/7))

2.5571E+01


One of the best ways to learn how to do things in python is by using google and reading about various functions. For example, I found a good description of the format() function at the [Geeks for geeks](https://www.geeksforgeeks.org/python-format-function/) website.

Python also allows for other math operations that you probably haven't used much, but are very useful for programming. The three examples we'll use are powers, modulo, and integer division. First, let's take a look at powers, which are almost as straightforward as addition, subtraction, multiplication, and division.

In [18]:
# what's 3^2?
3 ** 2

9

In [19]:
# what's 13^3.7?
13 ** 3.7

13230.930968301243

What about modulo and integer division? First, I should describe what those both are, since they are much less common than any of the math operations we've seen so far. When you first started learning division in school you probably didn't worry about fractions or decimals, but instead worried about remainders. Integer division and modulo operators are essentially this type of division. For example, instead of saying $\frac{22}{7}$=3.1429..., I can instead say that $\frac{22}{7}$=3r1 (3 with a remainder of 1). Integer division (the // operator) will give me the 3, while the modulo operator (represented by the % symbol) will give me the 1:

In [23]:
# how many times will 7 go into 22?
22 // 7

3

In [25]:
# what's the remainder when I divide 22 by 7?
22 % 7

1

In [29]:
# how many times will 3.1 go into 14.9?
print(14.9 // 3.1)

# what's the remainder when I divide 14.9 by 3.1?
print(14.9 % 3.1)

4.0
2.5


Those are pretty much all of the math operators built in to python. But what about other operations that you might want to use, such as square roots or logarithms? For those, we need to add an extra library of code into python. One of the more common and useful libraries is the numpy library, which contains implementations of many useful functions, such as natural logarithms, base-10 logarithms, and square roots. To add this to your notebook you use the "import" command:

In [30]:
import numpy as np

Now I can use all of numpy's features! For example, here are some useful functions in action:

In [34]:
# what's the square root of 13?
print(np.sqrt(13))

# what's the natural log of 152.7?
print(np.log(152.7))

# what's the base-10 log of 317.159?
print(np.log10(317.159))

3.60555127546
5.02847521222
2.50127703985


Finally, numpy also has common constants, such as $\pi$ and $e$, built in:

In [35]:
print("The first 10 digits of pi are {:.10}".format(np.pi))
print("The first 7 digits of e are {:.7}".format(np.e))
print("The square root of pi is {:.10}".format(np.sqrt(np.pi)))


The first 10 digits of pi are 3.141592654
The first 7 digits of e are 2.718282
The square root of pi is 1.772453851


In [41]:
np.e ** np.pi // np.pi ** np.e

1.0